### PR TITLE
Added more in-depth compiler checks

### PR DESCRIPTION
**Pull request by lulivi**
### Summary

Pull request solving #219 

### Details and comments

Also cleaned static analysis script documentation.

### CODE REVIEW

File: examples/connext_dds/dynamic_data_access_union_discriminator/c/CMakeLists.txt [hunk 1]
```diff
@@ -38,7 +38,10 @@ set_target_properties(dynamic_data_union_example_c
     PROPERTIES
         OUTPUT_NAME "dynamic_data_union_example")
 
-if(UNIX AND CMAKE_SYSTEM_NAME MATCHES "Linux")
+if(UNIX 
+    AND CMAKE_SYSTEM_NAME MATCHES "Linux" 
+    AND (CMAKE_C_COMPILER_ID MATCHES "GNU"
```
Comment (fgaranda) [inline examples/connext_dds/dynamic_data_access_union_discriminator/c/CMakeLists.txt]: In the C example you only need the C compiler.

File: examples/connext_dds/dynamic_data_access_union_discriminator/c++03/CMakeLists.txt [hunk 2]
```diff
@@ -33,7 +33,9 @@ set_target_properties(dynamic_data_union_example_cxx2
     PROPERTIES
         OUTPUT_NAME "dynamic_data_union_example")
 
-if(UNIX AND CMAKE_SYSTEM_NAME MATCHES "Linux")
+if(UNIX AND CMAKE_SYSTEM_NAME MATCHES "Linux" 
+    AND (CMAKE_C_COMPILER_ID MATCHES "GNU" 
+        OR CMAKE_CXX_COMPILER_ID MATCHES "GNU"))
```
Comment (fgaranda) [inline examples/connext_dds/dynamic_data_access_union_discriminator/c++03/CMakeLists.txt]: In C++ examples you only need C++ compilers

File: examples/connext_dds/dynamic_data_access_union_discriminator/c++03/CMakeLists.txt [hunk 3]
```diff
@@ -33,7 +33,9 @@ set_target_properties(dynamic_data_union_example_cxx2
     PROPERTIES
         OUTPUT_NAME "dynamic_data_union_example")
 
-if(UNIX AND CMAKE_SYSTEM_NAME MATCHES "Linux")
+if(UNIX AND CMAKE_SYSTEM_NAME MATCHES "Linux"
```
Comment (fgaranda) [inline examples/connext_dds/dynamic_data_access_union_discriminator/c++03/CMakeLists.txt]: Do you need UNIX if it's already "Linux"? Maybe you could simplify this.

File: examples/connext_dds/dynamic_data_access_union_discriminator/c++/CMakeLists.txt [hunk 4]
```diff
@@ -38,7 +38,9 @@ set_target_properties(dynamic_data_union_example_cxx
     PROPERTIES
         OUTPUT_NAME "dynamic_data_union_example")
 
-if(UNIX AND CMAKE_SYSTEM_NAME MATCHES "Linux")
+if(UNIX AND CMAKE_SYSTEM_NAME MATCHES "Linux" 
+    AND (CMAKE_C_COMPILER_ID MATCHES "GNU" 
+        OR CMAKE_CXX_COMPILER_ID MATCHES "GNU"))
     set_target_properties(dynamic_data_union_example_cxx
```
Comment (fgaranda) [inline examples/connext_dds/dynamic_data_access_union_discriminator/c++/CMakeLists.txt]: See comments below...

File: examples/connext_dds/dynamic_data_access_union_discriminator/c++03/CMakeLists.txt (hunk 3 above)
Comment (lulivi) [reply examples/connext_dds/dynamic_data_access_union_discriminator/c++03/CMakeLists.txt]: That's true.

**Pull request merged**